PowerNSX.psm1

```diff
@@ -16299,6 +16299,165 @@ function New-NsxEdgeRedistributionRule {
     end {}
 }
 
+function Invoke-NsxEdgeHaFailover {
+
+<#
+.SYNOPSIS
+Invokes a failover on a NSX Edge high availability pair.
+
+.DESCRIPTION
+NSX Edges can function with a HA option, which makes sure there is an active
+and standby NSX Edge, ready to take each others load if something
+happens to one.
+
+Starting NSX 6.2.3, there is a feature to manually fail over the NSX Edges.
+
+This cmdlet works with the Edge object returned by Get-NsxEdge.
+
+.EXAMPLE
+Invoke a failover of the NSX Edge named 'MyEdge':
+
+PS C:\> Get-NsxEdge -Name MyEdge | Invoke-NsxEdgeHaFailover
+
+Be specific in the Edge Node that will become the new primary Node:
+
+PS C:\> Get-NsxEdge -Name MyEdge | Invoke-NsxEdgeHaFailover -NewPrimaryNode 1
+
+#>
+
+  param (
+    [Parameter (Mandatory=$true,ValueFromPipeline=$true)]
+      #PowerNSX Edge Object as retrieved with Get-NsxEdge
+      [ValidateScript({ Validate-Edge $_ })]
+      [System.Xml.XmlElement]$Edge,
+    [Parameter (Mandatory=$false)]
+      #Which Edge Node ID to make the new primary node
+      [ValidateRange(0,1)]
+      [int]$NewPrimaryNode=-1,
+    [Parameter (Mandatory=$False)]
+      #PowerNSX Connection object
+      [ValidateNotNullOrEmpty()]
+      [PSCustomObject]$Connection=$defaultNSXConnection
+  )
+
+  begin {}
+
+  process
+  {
```

Comment:
No need for a new line for opening curly braces "{". Just stick them on the end of the preceding line.